### ISSUE TITLE
3 missing pypy builds for 1.7.0

### ISSUE THREAD

**Issue by h-vetinari** [conda-forge/scipy-feedstock#173]
See discussion in #169, culminating in [this comment](https://github.com/conda-forge/scipy-feedstock/pull/169#issuecomment-869601570): 

> That means 1.7.0 has now been built for all but the following three variants:
>   * pypy37 on linux / x64
>   * pypy37 on linux / ppc
>   * pypy37 on osx / x64

**Comment by mattip** [comment]
I think the PyPy crash is due to a huge stack allocation when entering a function, cython/cython#4272. I don't understand what the difference is between CPython and PyPy.

**Comment by h-vetinari** [comment]
The PPC build will need some encouragement (=restarts), keeping this open until then.

**Comment by mattip** [comment]
I think windows64 pypy migration is missing as well?

**Comment by h-vetinari** [comment]
Yes indeed, but I wasn't counting that since the migrator still needs to chew through a bunch of dependencies first

**Comment by mattip** [comment]
Is the migrator still active? At https://conda-forge.org/status/ I see it is listed under "Recently closed migrations" as "pypy37-windows closed at 2021-07-07"

**Comment by h-vetinari** [comment]
See my comment [here](https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/1668#issuecomment-882343614)

**Comment by h-vetinari** [comment]
I cannot get the pypy build on PPC to pass despite >5 restarts. Perhaps it needs some of the recent fixes on the pypy-feedstock (and PPC needing manual rebuilds)

**Comment by mattip** [comment]
I personally would be happy to drop ppc and s390x support for PyPy on conda forge. At least until we get a little more experience around the use case for PyPy on x86/arm64.

**Comment by isuruf** [comment]
travis is supposed to run only a fraction as in https://github.com/conda-forge/scipy-feedstock/blob/master/recipe/run_test.py#L19-L21. Not sure why that doesn't get triggered.

**Comment by h-vetinari** [comment]
@isuruf, it's not a question of runtime. The build regularly fails with:
```
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.

Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received
```

It could just be extreme flakiness on the side of travis, but it could equally be a hang in the pypy builds. If you could rebuild the current pypy-recipe for aarch, we could test that hypothesis. Otherwise, I tend to agree with Matti; PPC is already <0.1% of downlods, and pypy-on-PPC is <0.01% of downloads, cf. https://github.com/conda-forge/scipy-feedstock/issues/171

**Comment by rgommers** [comment]
+1 for disabling PyPy on PPC, too niche to be worth even this current effort.

**Comment by isuruf** [comment]
Problem with disabling now is that downstream packages will need to disable that platform manually which I want to avoid.

Maybe we should try cross compiling so that we can move away from Travis-CI. Travis-CI has been extremely flaky lately.
I'll see if I can get cross-python to support PyPy. Might need changes in PyPy.

**Comment by h-vetinari** [comment]
I've tried cross-compiling in Azure, but that creates ~2000 test suite errors in the test suite (not just for PyPy), see [here](https://github.com/conda-forge/scipy-feedstock/pull/169#issuecomment-860688710).

**Comment by h-vetinari** [comment]
> Maybe we should try cross compiling so that we can move away from Travis-CI.

See #179

**Comment by h-vetinari** [comment]
Unexpectedly, cross-compilation on ppc now works without issue. This will be solved as soon as #182 is merged.
